src/core/api/__tests__/public_api.test.ts

```diff
@@ -0,0 +1,571 @@
+        expect(player.getManualVideoBitrate()).to.equal(-1);
+      });
+
+      it("should update manual video bitrate when calling setVideoBitrate", () => {
+        const oldManual = player.getManualVideoBitrate();
+
+        player.setVideoBitrate(84);
+        expect(player.getManualVideoBitrate()).to.equal(84);
+
+        player.setVideoBitrate(-1);
+        expect(player.getManualVideoBitrate()).to.equal(-1);
+
+        player.setVideoBitrate(0);
+        expect(player.getManualVideoBitrate()).to.equal(0);
+
+        player.setVideoBitrate(oldManual);
+        expect(player.getManualVideoBitrate()).to.equal(oldManual);
+      });
+    });
+
+    describe("setMaxVideoBitrate/getMaxVideoBitrate", () => {
+      /* tslint:disable:max-line-length */
+      it("should update the maximum video bitrate when calling setMaxVideoBitrate by default", () => {
+      /* tslint:enable:max-line-length */
+        const oldMax = player.getManualVideoBitrate();
+
+        expect(player.setMaxVideoBitrate(Infinity)).to.equal(undefined);
+        expect(player.getMaxVideoBitrate()).to.equal(Infinity);
+
+        expect(player.setMaxVideoBitrate(500)).to.equal(undefined);
+        expect(player.getMaxVideoBitrate()).to.equal(500);
+
+        expect(player.setMaxVideoBitrate(3)).to.equal(undefined);
+        expect(player.getMaxVideoBitrate()).to.equal(3);
+
+        expect(player.setMaxVideoBitrate(Infinity)).to.equal(undefined);
+        expect(player.getMaxVideoBitrate()).to.equal(Infinity);
+
+        expect(player.setMaxVideoBitrate(oldMax)).to.equal(undefined);
+        expect(player.getMaxVideoBitrate()).to.equal(oldMax);
+      });
+    });
+
+    describe("setMaxAudioBitrate/getMaxAudioBitrate", () => {
+      /* tslint:disable:max-line-length */
+      it("should update the maximum audio bitrate when calling setMaxAudioBitrate by default", () => {
+      /* tslint:enable:max-line-length */
+        const oldMax = player.getManualAudioBitrate();
+
+        expect(player.setMaxAudioBitrate(Infinity)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(Infinity);
+
+        expect(player.setMaxAudioBitrate(500)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(500);
+
+        expect(player.setMaxAudioBitrate(3)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(3);
+
+        expect(player.setMaxAudioBitrate(Infinity)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(Infinity);
+
+        expect(player.setMaxAudioBitrate(oldMax)).to.equal(undefined);
+        expect(player.getMaxAudioBitrate()).to.equal(oldMax);
+      });
+    });
+
+    describe("getMaxBufferBehind/setMaxBufferBehind", () => {
+      /* tslint:disable:max-line-length */
+      it("should update the max buffer behind through setMaxBufferBehind by default", () => {
+      /* tslint:enable:max-line-length */
+        expect(player.setMaxBufferBehind(50)).to.equal(undefined);
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Remove this use of the output from "setMaxBufferBehind"; "setMaxBufferBehind" doesn't return anything. ([typescript:S3699](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS3699))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=414&issues=AWgOMLyH_MPaBDn0sxTQ&open=AWgOMLyH_MPaBDn0sxTQ)

[ISSUE_KEY]: AWgOMLyH_MPaBDn0sxTQ